### ISSUE TITLE
[Bug] Incorrect report of partition size to spark driver

### ISSUE THREAD

**Issue by zuston** [apache/incubator-uniffle#589]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

In shuffle writer side, the writer will report its partitions size to driver when invoking the `stop` method in `RssShuffleWriter`. But it looks writer will report the every partition compressed size, which will cause incorrect optimization by AQE.

### Affects Version(s)

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by zuston** [comment]
PTAL @advancedxy @jerqi @xianjingfeng

**Comment by jerqi** [comment]
I look at Spark code. Origin Spark also use compressed data length, too.

**Comment by zuston** [comment]
> I look at Spark code. Origin Spark also use compressed data length, too.

You are right. But I think this is wrong, and spark also has similar problem.

**Comment by jerqi** [comment]
> > I look at Spark code. Origin Spark also use compressed data length, too.
> 
> You are right. But I think this is wrong, and spark also has similar problem.

We should keep same behaviour with Spark. If you think it's a bug, you should modify it in the Spark first.

**Comment by zuston** [comment]
> Do you have a concrete case that AQE doesn't work as expected? For that case, is vanilla spark worked as expected?

I haven't do some experiments to confirm this. Maybe dedicated test case is enough.

> I believe you could file a jira to Spark and discuss the possibility of add an additional field of uncompressed size of shuffle data with the Spark community.

Yes.
